client/web/web.go

```diff
@@ -202,6 +206,23 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 }
 
 func (s *Server) serve(w http.ResponseWriter, r *http.Request) {
+	// serve HTTP 204 on /ok requests as connectivity check
+	if r.Method == httpm.GET && r.URL.Path == "/ok" {
+		w.WriteHeader(http.StatusNoContent)
+		return
+	}
+
+	if s.mode == ManageServerMode {
+		// In manage mode, requests must be sent directly to the bare Tailscale IP address.
+		// If a request comes in on any other hostname, redirect.
+		if s.requireTailscaleIP(w, r) {
+			return
```

Comment:
nit: maybe a little comment here that they were redirected

```
return // user was redirected
```

Reply:
done

Reply:
Yea I was thinking about that a little too. I guess you could have the function just return `hostIsTailscaleIP` or something like that and do the redirect from here. Although you're using the ipv4 addr from the ip loop so you'd probably have to pass that too, which might just be messier.